app-desktop/build.gradle.kts

```diff
@@ -1,45 +1,129 @@
     alias(libs.plugins.jetbrainsCompose)
     alias(libs.plugins.composeCompiler)
     alias(libs.plugins.composeHotReload)
     id("droidkaigi.primitive.metro")
     id("droidkaigi.primitive.aboutlibraries")
     id("droidkaigi.primitive.spotless")
+    id("droidkaigi.primitive.kmp.compose.resources")
+}
+
+compose {
+    resources {
+        nameOfResClass = "AppDesktopRes"
+    }
 }
 
 val aboutLibrariesTargetDir = "${layout.buildDirectory.get().asFile.path}/generated/aboutlibraries"
 
 kotlin {
+    jvm("desktop")
+
+    // JDK Version 21 is required to create the distribution materials.
+    jvmToolchain(21)
+
     compilerOptions {
         freeCompilerArgs.add("-Xcontext-parameters")
     }
 
     sourceSets {
-        main {
+        val commonMain by getting {
+            dependencies {
+                implementation(projects.appShared)
+                implementation(projects.core.data)
+                implementation(compose.runtime)
+                implementation(compose.foundation)
+                implementation(compose.ui)
+            }
+        }
+        val desktopMain by getting {
+            dependencies {
+                implementation(compose.desktop.currentOs)
+                implementation(libs.navigationeventDesktop)
+                implementation(libs.lifecycleRuntimeCompose)
+                implementation(compose.uiUtil)
+            }
             resources.srcDir(aboutLibrariesTargetDir)
         }
     }
 }
 
 compose.desktop.application.mainClass = "io.github.droidkaigi.confsched.MainKt"
 
-dependencies {
-    implementation(projects.appShared)
-    implementation(projects.core.data)
-    implementation(compose.desktop.currentOs)
-    implementation(compose.foundation)
-    implementation(compose.runtime)
-    implementation(libs.navigationeventDesktop)
-    implementation(libs.lifecycleRuntimeCompose)
-    implementation(compose.ui)
-    implementation(compose.uiUtil)
-}
-
 aboutLibraries.export {
     outputFile.set(file("${aboutLibrariesTargetDir}/licenses.json"))
 }
 
-tasks.named("processResources") {
-    dependsOn("exportLibraryDefinitions")
+compose.desktop {
+    application {
+        nativeDistributions {
+            targetFormats(
+                // for MacOS
+                org.jetbrains.compose.desktop.application.dsl.TargetFormat.Dmg,
+                // for Windows
+                org.jetbrains.compose.desktop.application.dsl.TargetFormat.Msi,
+                org.jetbrains.compose.desktop.application.dsl.TargetFormat.Exe,
+                // for Linux
+                org.jetbrains.compose.desktop.application.dsl.TargetFormat.Deb,
+                org.jetbrains.compose.desktop.application.dsl.TargetFormat.Rpm
+            )
+
+            macOS {
+                iconFile.set(project.file("src/desktopMain/resources/DroidKaigi2025.icns"))
+            }
+            windows {
+                iconFile.set(project.file("src/desktopMain/resources/DroidKaigi2025.ico"))
```

Comment:
We created ico using the following commands and checked whether icons of each size were included.
The magic command is from ImageMagick.

```shell
magick icon_1024x1024.png -define icon:auto-resize=16,32,48,256 app.ico

or

magick icon_1024x1024.png \
  \( -clone 0 -resize 256x256 \) \
  \( -clone 0 -resize 48x48 \) \
  \( -clone 0 -resize 32x32 \) \
  \( -clone 0 -resize 16x16 \) \
  -delete 0 -alpha on -colorspace sRGB app.ico

identify -format "%f %wx%h\n" app.ico
```